relecov_tools/mail.py

```diff
@@ -1,67 +1,104 @@
-AUTHOR: Guillermo J. Gorines Cordero
-MAIL: guillermo.gorines@urjc.es
-VERSION: 0
-CREATED: 7-3-2022
-REVISED: 7-3-2022
-REVISED BY: guillermo.gorines@urjc.es
-DESCRIPTION:
+import logging
+import os
+import smtplib
+import relecov_tools.utils
+from email.mime.multipart import MIMEMultipart
+from email.mime.text import MIMEText
+from email.mime.base import MIMEBase
+from email import encoders
+from jinja2 import Environment, FileSystemLoader
 
-    Includes the Email, and its associated methods.
 
-REQUIREMENTS:
-    -Python
+log = logging.getLogger(__name__)
 
-TO DO:
 
+class EmailSender:
+    def __init__(self, config):
+        self.config = config
+        self.template_path = self.config.get("delivery_template_path_file")
+        self.yaml_cred_path = self.config.get("yaml_cred_path")
 
-================================================================
-END_OF_HEADER
-================================================================
-"""
+        if not self.config:
+            raise ValueError("Configuration not loaded correctly.")
 
-# Imports
-import smtplib
+        if not os.path.exists(self.template_path):
+            raise FileNotFoundError(
+                f"The template file could not be found in path {self.template_path}.")
 
-from email.mime.multipart import MIMEMultipart
-from email.mime.text import MIMEText
+    def get_institution_info(
+        self, institution_code, institutions_file="institutions.json"
+    ):
+        """
+        Load the institution's information from the JSON file.
+        """
+        institutions_file = self.config.get(
+            "institutions_guide_path", "institutions_guide.json"
+        )
+        institutions_data = relecov_tools.utils.read_json_file(institutions_file)
+
+        if institutions_data and institution_code in institutions_data:
+            return institutions_data[institution_code]
+        else:
+            print(f"No information found for code {institution_code}")
+            return None
+
+    def render_email_template(self, additional_info="",
+                              invalid_count= None,
+                              submitting_institution_code=None):
+        
+        institution_info = self.get_institution_info(submitting_institution_code)
+        if not institution_info:
+            print("Error: The information could not be obtained from the institution.")
+            return None, None
+
+        institution_name = institution_info["institution_name"]
+        email_receiver = institution_info["email_receiver"]
+
+        env = Environment(loader=FileSystemLoader(os.path.dirname(self.template_path)))
+        template = env.get_template(os.path.basename(self.template_path))
+
+        email_template = template.render(
+            submitting_institution=institution_name,
+            invalid_count=invalid_count,
+            additional_info=additional_info,
+        )
+
+        return email_template, email_receiver, institution_name
```

Comment:
I think you should only return `email_template` here. I think it makes more sense that if you want to get `email_receiver` and "`institution_name` you should call `EmailSender.get_institution_info()` and extract them from the returned dict. Not only that but reading jinja2's documentation i think that you can directly pass a dictionary to render so it would look like this:
```suggestion

        template_vars_dict = institution_info.copy() 
        template_vars_dict["invalid_count"] = invalid_count
        template_vars_dict["additional_info"] = additional_info

        env = Environment(loader=FileSystemLoader(os.path.dirname(self.template_path)))
        template = env.get_template(os.path.basename(self.template_path))

        email_template = template.render(template_vars_dict)

        return email_template
```
Give it a try, I think this should work